libraries/botbuilder-dialogs-adaptive-runtime/src/index.ts

```diff
@@ -411,11 +411,16 @@ export async function getRuntimeServices(
     // Resolve configuration
     let configuration: Configuration;
     if (typeof configurationOrSettingsDirectory === 'string') {
-        configuration = new Configuration()
-            .argv()
-            .env()
-            .file(path.join(configurationOrSettingsDirectory, 'appsettings.Development.json'))
-            .file(path.join(configurationOrSettingsDirectory, 'appsettings.json'));
+        configuration = new Configuration().argv().env();
+
+        const files = ['appsettings.json'];
+
+        const { NODE_ENV: nodeEnv } = process.env;
+        if (nodeEnv) {
+            files.unshift(`appsettings.${nodeEnv}.json`);
+        }
```

Comment:
Somehow my comment was lost... The norm in .NET is to use `appsettings.Development.json` which means the SDK should standardize on an expected casing pattern. Perhaps capitalize the value of `NODE_ENV`? `NODE_ENV` is normally a lowercase value, e.g. [`production`](https://expressjs.com/en/advanced/best-practice-performance.html#set-node_env-to-production) and [`development`](https://github.com/expressjs/express/blob/508936853a6e311099c9985d4c11a4b1b8f6af07/lib/application.js#L71).

Do the generators create the same filenames (`appsettings.json` and `appsettings.Development.json`) for both .NET and JS/TS bots?

Reply:
Generators don't create environment-specific settings files, at least not at the moment. To me, using `process.env.NODE_ENV` directly is a more predictable behavior from the JS perspective. Once someone generates a project, the selected language is fixed. In this case, I think it's better to be consistent with JS expectations rather than trying to force .NET patterns. Doing zero transformations to the `NODE_ENV` environment variable gives users the most flexibility, and they don't have to try to understand a potentially fluid set of rules.

Reply:
That works for me! I asked because it's important to have these discussions to make sure we as a team move in the right direction for our customers.